r/tests/testthat/test-dataset-write.R

```diff
@@ -506,6 +506,47 @@ test_that("Max partitions fails with non-integer values and less than required p
   )
 })
 
+test_that("max_rows_per_group is adjusted if at odds with max_rows_per_file", {
+  skip_if_not_available("parquet")
+  df <- tibble::tibble(
+    int = 1:10,
+    dbl = as.numeric(1:10),
+    lgl = rep(c(TRUE, FALSE, NA, TRUE, FALSE), 2),
+    chr = letters[1:10],
+  )
+  dst_dir <- make_temp_dir()
+
+  # max_rows_per_group unset => pass
+  expect_silent(
+    write_dataset(df, dst_dir, max_rows_per_file = 5)
+  )
+
+  expect_equal(
+    {
+      write_dataset(df, dst_dir, max_rows_per_file = 5)
+      list.files(dst_dir, "part-") %>%
+        length()
+    },
+    2
+  )
```

Comment:
Great attention to detail, but I think we can remove this test, as it's a little out of scope of this ticket - it's basically testing that `max_rows_per_file` works as intended (which should be tested in the C++ layer anyway), rather than the specific thing this ticket addresses (adjusting the behaviour when the user specified `max_rows_per_file` at odds with `max_rows_per_group`).

Reply:
Agree, I was unsure writing it and you worded my doubts clearly, thanks!